### DISCUSSION TITLE
Cannot update languages - could not find an editable version for 'IOS'

### DISCUSSION THREAD

**Discussion by sneko** [fastlane/fastlane#19074] [Q&A]
### New Issue Checklist

- [ ] Updated fastlane to the latest version
- [ ] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [ ] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [ ] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description

Hi, I got an issue that is exactly this one https://github.com/fastlane/fastlane/issues/17905 but unfortunately it has been closed without any solution/explanation. (cc @brandtdaniels @tomaszpieczykolan if you have some thoughts on it I would appreciate).

When upload to the main App Store (not testflight) I get this:
```
[20:11:48]: Detected languages: ["default"]
[20:11:48]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 4)
[20:11:59]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 3)
[20:12:09]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 2)
[20:12:20]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 1)
[20:12:30]: Cannot find edit app store version... Retrying after 10 seconds (remaining: 0)
```

And the final error is:
```
[20:12:40]: Cannot update languages - could not find an editable version for 'IOS'
```

On the Fastlane script I have this:
```
  prepare_deploy(
    scheme: 'Production',
    remote_build_number: 4,
  )

  build_app(
    scheme: 'Production',
    export_xcargs: "-allowProvisioningUpdates",
    output_directory: buildDirectory,
  )
  upload_to_app_store(
    force: true, # Skip HTMl report verification
    skip_screenshots: true,
    skip_app_version_update: true,
    ipa: buildDirectory + 'Production.ipa',
    release_notes: {
      "default": get_changelog,
    },
    submit_for_review: true,
  )
```

_Note that I already have a version published on the store that is fully reviewed, and I made sure the `remote_build_number` is different. By the way at start I made it incrementing automatically but for the test to debug the issue I set it manually._

Since there is a retry system, I guess that's not something local it tries to find. But what could be missing in my App Store dashboard 🤔  ?

Thank you in advance,

### Environment

<details>
  <pre><details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 11.4                                        |
| Ruby                        | 3.0.0                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.30.1 (Apple Git-130)          |
| Installation Source         | ~/.rvm/gems/ruby-3.0.0/bin/fastlane         |
| Host                        | macOS 11.4 (20F71)                          |
| Ruby Lib Dir                | ~/.rvm/rubies/ruby-3.0.0/lib                |
| OpenSSL Version             | OpenSSL 1.1.1k  25 Mar 2021                 |
| Is contained                | false                                       |
| Is homebrew                 | false                                       |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 12.5.1                                      |
| Swift Version               | 5.4.2                                       |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

**No Fastfile found**

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status      |
| -------- | ------- | ------------------ |
| fastlane | 2.185.0 | 🚫 Update available |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                             | Version      |
| ------------------------------- | ------------ |
| did_you_mean                    | 1.5.0        |
| executable-hooks                | 1.6.1        |
| bundler-unload                  | 1.0.2        |
| rubygems-bundler                | 1.4.5        |
| bundler                         | 2.2.3        |
| pathname                        | 0.1.0        |
| atomos                          | 0.1.3        |
| CFPropertyList                  | 3.0.3        |
| claide                          | 1.0.3        |
| colored2                        | 3.1.2        |
| nanaimo                         | 0.3.0        |
| xcodeproj                       | 1.19.0       |
| rouge                           | 2.0.7        |
| xcpretty                        | 0.3.0        |
| terminal-notifier               | 2.0.0        |
| unicode-display_width           | 1.7.0        |
| terminal-table                  | 1.8.0        |
| plist                           | 3.6.0        |
| public_suffix                   | 4.0.6        |
| addressable                     | 2.7.0        |
| multipart-post                  | 2.0.0        |
| word_wrap                       | 1.0.0        |
| tty-screen                      | 0.8.1        |
| tty-cursor                      | 0.7.1        |
| tty-spinner                     | 0.9.3        |
| artifactory                     | 3.0.15       |
| babosa                          | 1.0.4        |
| colored                         | 1.2          |
| highline                        | 2.0.3        |
| commander                       | 4.6.0        |
| faraday-em_http                 | 1.0.0        |
| faraday-em_synchrony            | 1.0.0        |
| faraday-excon                   | 1.1.0        |
| faraday-net_http                | 1.0.1        |
| faraday-net_http_persistent     | 1.1.0        |
| ruby2_keywords                  | 0.0.4        |
| faraday                         | 1.4.2        |
| faraday-cookie_jar              | 0.0.7        |
| faraday_middleware              | 1.0.0        |
| gh_inspector                    | 1.1.3        |
| json                            | 2.5.1        |
| mini_magick                     | 4.11.0       |
| naturally                       | 2.2.1        |
| rubyzip                         | 2.3.0        |
| security                        | 0.1.3        |
| xcpretty-travis-formatter       | 1.0.1        |
| dotenv                          | 2.7.6        |
| simctl                          | 1.6.8        |
| jwt                             | 2.2.3        |
| uber                            | 0.1.0        |
| declarative                     | 0.0.20       |
| trailblazer-option              | 0.1.1        |
| representable                   | 3.1.1        |
| retriable                       | 3.1.2        |
| mini_mime                       | 1.1.0        |
| multi_json                      | 1.15.0       |
| signet                          | 0.15.0       |
| memoist                         | 0.16.2       |
| os                              | 1.1.1        |
| googleauth                      | 0.16.2       |
| httpclient                      | 2.8.3        |
| webrick                         | 1.7.0        |
| google-apis-core                | 0.3.0        |
| google-apis-playcustomapp_v1    | 0.3.0        |
| google-cloud-env                | 1.5.0        |
| google-cloud-errors             | 1.1.0        |
| google-cloud-core               | 1.6.0        |
| google-apis-iamcredentials_v1   | 0.4.0        |
| google-apis-storage_v1          | 0.4.0        |
| rake                            | 13.0.3       |
| digest-crc                      | 0.6.3        |
| google-cloud-storage            | 1.31.1       |
| emoji_regex                     | 3.2.2        |
| tsort                           | 0.1.0        |
| set                             | 1.0.1        |
| uri                             | 0.10.1       |
| excon                           | 0.82.0       |
| unf_ext                         | 0.0.7.7      |
| unf                             | 0.1.4        |
| domain_name                     | 0.5.20190701 |
| http-cookie                     | 1.0.4        |
| fastimage                       | 2.2.4        |
| rexml                           | 3.2.5        |
| google-apis-androidpublisher_v3 | 0.5.0        |
| aws-eventstream                 | 1.1.1        |
| aws-sigv4                       | 1.2.3        |
| aws-partitions                  | 1.466.0      |
| jmespath                        | 1.4.0        |
| aws-sdk-core                    | 3.114.2      |
| aws-sdk-kms                     | 1.43.0       |
| aws-sdk-s3                      | 1.96.1       |
| forwardable                     | 1.3.2        |
| logger                          | 1.4.3        |
| shellwords                      | 0.1.0        |
| cgi                             | 0.2.0        |
| date                            | 3.1.0        |
| timeout                         | 0.1.1        |
| stringio                        | 3.0.0        |
| openssl                         | 2.2.0        |
| io-nonblock                     | 0.1.0        |
| ipaddr                          | 1.2.2        |
| io-wait                         | 0.1.0        |
| zlib                            | 1.1.0        |
| resolv                          | 0.2.0        |
| securerandom                    | 0.1.0        |
| net-protocol                    | 0.1.0        |
| digest                          | 3.0.0        |
| net-smtp                        | 0.2.1        |
| ostruct                         | 0.3.1        |
| english                         | 0.7.1        |
| erb                             | 2.2.0        |
| strscan                         | 3.0.0        |
| optparse                        | 0.1.0        |
| abbrev                          | 0.1.0        |
| io-console                      | 0.5.6        |
| tempfile                        | 0.1.1        |
| delegate                        | 0.2.0        |
| fileutils                       | 1.5.0        |
| tmpdir                          | 0.1.1        |
| base64                          | 0.1.0        |
| time                            | 0.1.0        |
| singleton                       | 0.1.1        |
| open3                           | 0.1.1        |
| nkf                             | 0.1.0        |
| prettyprint                     | 0.1.0        |
| pp                              | 0.1.0        |
| find                            | 0.1.0        |
| yaml                            | 0.1.1        |
| psych                           | 3.3.0        |
| open-uri                        | 0.1.0        |
| mutex_m                         | 0.1.1        |
</details>

*generated on:* **2021-07-10**
</details></pre>
</details>

EDIT: cannot retry for a few days but I'm wondering if it could come from the:
```
skip_app_version_update | Don’t create or update the app version that is being prepared for submission
```
I set to `true`

**Comment by lucgrabowski** [comment]
Hi @sneko  I think that you've found a reason for it. You currently see on Appstore connect one live version of the app and there is no edit version in Prepare for submission state. When you set skip_app_version_update to true, new edit version cannot be created. If you set it to false, all should be fine.

**Comment by sneko** [comment]
Oh great @lucgrabowski ! Thank you for the answer 🎆 

I won't be able to test it within 8 days (no more Macbook ^^), but will close the issue ASAP after succeeding to upload a new version :)

Thanks!

EDIT: maybe the error could be more explicit?

**Comment by crazymanish** [comment]
Converted this to discussion so it might be helpful for others too! 😇

**Comment by mariabernis-tw** [comment]
I'm having the same issue and I have `skip_app_version_update` set to false . The **upload_to_app_store** action fails with the mentioned message IF there is no new release created already in App Store Connect (one in yellow with status "Prepare for Submission"). Looks like is not able to create a new release. 

This is how we have configured the action in our Fastfile: we upload the build in a previous step and we call **upload_to_app_store** to create a new release for that latest build and submit it for review. 

```ruby
upload_to_app_store(
        api_key_path: ios_app_store_connect_api_key_path,
        team_id: ios_app_store_connect_team_id,
        app_identifier: app_identifier,
        app_icon: ios_app_icon_path,
        price_tier: 0,
        submit_for_review: true,
        reject_if_possible: true,
        force: true,
        automatic_release: false,
        skip_app_version_update: false,
        overwrite_screenshots: true,
        skip_binary_upload: true,
        ignore_language_directory_validation: true,
        precheck_include_in_app_purchases: false,

        submission_information: {
            add_id_info_uses_idfa: false,
            content_rights_contains_third_party_content: false,
        },
        metadata_path: ios_metadata_path,
        screenshots_path: ios_screenshots_path,
        app_rating_config_path: ios_app_rating_path,
        release_notes: {
            'default' => release_notes_en,
            'en-GB' => release_notes_en
        },
        app_review_information: {
            email_address: ios_review_info["contact_email"],
            first_name: ios_review_info["contact_first_name"],
            last_name: ios_review_info["contact_last_name"],
            phone_number: ios_review_info["contact_phone"],
            notes: ios_review_info["notes"],
            demo_user: ios_review_info["demo_account_name"],
            demo_password: ios_review_info["demo_account_password"]
        },
        app_review_attachment_file: ios_review_nfc_video_path
    )
```

**Reply by mariabernis-tw** [reply]
For anyone facing this error, we found out what was missing on our side: we needed to specify `app_version` or it won't know what release to create. 

I agree with @sneko that the error and logs could be better to indicate there's a step "Checking if release already exist" and "Creating new release", but just a suggestion :)
